stm32f0_hal/src/uart.rs

```diff
@@ -29,164 +30,295 @@ pub enum Parity {
     Odd,
 }
 
-pub fn setup<F>(baudrate: u32, nbits: NBits, nbstopbits: StopBits, parity: Parity, mut f: F)
```

Comment:
I test the baudrate by setting it to 115200 but I get a 19323 baudrate output...